### PR TITLE
uid: make task match warning explicit

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Make implicit globs explicit so that warning messages are more
  informative.
* E.G. `cylc hold foo` should produce the warning
  `no tasks matching foo/*`.

### CODE REVIEW

**Comment by MetRonnie** [comment]
Just noticed something (not currently touched by this PR, but probably worth bunging in a fix) - shouldn't there be a `continue` after this warning?
https://github.com/cylc/cylc-flow/blob/1f6364d1a02187850e1da5a9c9c222f5992e066f/cylc/flow/task_pool.py#L1619-L1622

File: cylc/flow/task_pool.py [hunk 1]
```diff
@@ -1677,25 +1675,33 @@ def match_taskdefs(
         """
         n_warnings = 0
         task_items: Dict[Tuple[str, 'PointBase'], 'TaskDef'] = {}
-        for item in items:
-            tokens = Tokens(item, relative=True)
+        for id_ in ids:
+            try:
+                tokens = Tokens(id_, relative=True)
+            except ValueError:
+                LOG.warning(f'Invalid task ID: {id_}')
```
Comment (MetRonnie) [inline cylc/flow/task_pool.py:1682]: As with https://github.com/cylc/cylc-flow/pull/4597#issuecomment-1017699862
```suggestion
                LOG.warning(f'Invalid task ID: {id_}')
                continue
```

File: cylc/flow/task_pool.py [hunk 2]
```diff
@@ -1629,23 +1629,21 @@ def match_future_tasks(
                 LOG.warning(f"No active tasks matching: {id_}")
                 unmatched_tasks.append(id_)
                 continue
+            if not tokens['task']:
+                # make task globs explicit to make warnings clearer
+                tokens['task'] = '*'
```
Comment (MetRonnie) [inline cylc/flow/task_pool.py:1634]: I think if this is moved to above the previous `if` condition, then the `No active tasks in the family` message below would be fine

Comment (oliver-sanders) [reply cylc/flow/task_pool.py:1634]: The "in family" bit is misnomer, for the reported example the task is unspecified i.e. "*", so reporting it this way is confusing. Clearer to say "no tasks match ...".

File: cylc/flow/task_pool.py (hunk 1 above)
Comment (oliver-sanders) [reply cylc/flow/task_pool.py:1682]: Will take a look

Comment (oliver-sanders) [reply cylc/flow/task_pool.py:1682]: Yep, that's a glaring bug.

File: cylc/flow/task_pool.py [hunk 3]
```diff
@@ -1629,23 +1630,21 @@ def match_future_tasks(
...
                     LOG.warning(
-                        f"No active tasks in the family '{name_str}' "
-                        f"matching: {id_}"
+                        f"No active tasks matching: {tokens.relative_id}"
                     )
                 else:
                     LOG.warning(self.ERR_TMPL_NO_TASKID_MATCH.format(name_str))
                 unmatched_tasks.append(id_)
                 continue
-            if point_str is None:
```
Comment (oliver-sanders) [inline cylc/flow/task_pool.py:1645]: This should have been removed with the UID branch as point_str cannot be None.

File: cylc/flow/task_pool.py (hunk 2 above)
Comment (MetRonnie) [reply cylc/flow/task_pool.py:1635]: But if this is moved above this bit:
https://github.com/cylc/cylc-flow/blob/1f6364d1a02187850e1da5a9c9c222f5992e066f/cylc/flow/task_pool.py#L1623-L1628

then the only way of reaching here:
https://github.com/cylc/cylc-flow/blob/1f6364d1a02187850e1da5a9c9c222f5992e066f/cylc/flow/task_pool.py#L1637-L1639

is as the comment says, if the name is a family name. So it wouldn't be a misnomer

**Comment by oliver-sanders** [comment]
Ok, how about this?

**Comment by MetRonnie** [comment]
Nice, it keeps the ability to distinguish family names that have no active tasks.

Just a couple of test failures due to error message wording
```
FAILED tests/integration/test_task_pool.py::test_hold_tasks[Family names hold active tasks only]
FAILED tests/integration/test_task_pool.py::test_hold_tasks[Specifying task state works for active tasks, not future tasks]
```

**Pull request merged**